caas/kubernetes/provider/application/application.go

```diff
@@ -1338,6 +1346,32 @@ func (a *app) applicationPodSpec(config caas.ApplicationConfig) (*corev1.PodSpec
 				Name:  "PEBBLE_SOCKET",
 				Value: "/charm/container/pebble.socket",
 			}},
+			LivenessProbe: &corev1.Probe{
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Path: "/v1/health?level=alive",
+						Port: intstr.FromInt(containerPebblePortStart + i),
+					},
+				},
+				InitialDelaySeconds: containerProbeInitialDelay,
+				TimeoutSeconds:      containerProbeTimeout,
+				PeriodSeconds:       containerProbePeriod,
+				SuccessThreshold:    containerProbeSuccess,
+				FailureThreshold:    containerProbeFailure,
+			},
+			ReadinessProbe: &corev1.Probe{
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Path: "/v1/health?level=ready",
+						Port: intstr.FromInt(containerPebblePortStart + i),
+					},
+				},
+				InitialDelaySeconds: containerProbeInitialDelay,
+				TimeoutSeconds:      containerProbeTimeout,
+				PeriodSeconds:       containerProbePeriod,
+				SuccessThreshold:    containerProbeSuccess,
+				FailureThreshold:    containerProbeFailure,
+			},
```

Comment:
This reminds me we need to finish the caasprober worker to actually pull from the local state of the uniter. Currently liveness, readiness and startup probes just check if the unit agent is alive.